lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -174,7 +174,10 @@ def relationship_value_for(serializer, options = {})
         end
 
         def relationships_for(serializer)
-          serializer.associations.each_with_object({}) do |association, hash|
+          resource_type = resource_identifier_type_for(serializer)
+          requested_associations = fieldset.try(:fields_for, resource_type) || '*'
```

Comment:
the situation you just described with fieldset.
It's important to have docs like this when someone new starts working on the json api adapter.